src/Codeception/Module/WebDriver.php

```diff
@@ -511,11 +507,13 @@ public function _getCurrentUri()
 
     public function _saveScreenshot($filename)
     {
-        if ($this->webDriver !== null) {
+        try {
+            if ($this->webDriver === null) {
+                throw new \Exception("WebDriver::_saveScreenshot method has been called when webDriver is not set");
```

Comment:
I agree with you but I didn't know how to do that. I don't want to repeat the error.

```php
try {
if ($this->webDriver === null) {
throw new \Exception("WebDriver::_saveScreenshot method has been called when webDriver is not set");
}
$this->webDriver->takeScreenshot($filename);
 } catch (\Exception $e) {
 $this->debug("Unable to retrieve screenshot from Selenium : ".$e->getMessage());
}
```

or I can do that if you prefer
```php
if ($this->webDriver === null) {
$this->debug("WebDriver::_saveScreenshot method has been called when webDriver is not set");
}

try {
$this->webDriver->takeScreenshot($filename);
 } catch (\Exception $e) {
 $this->debug("Unable to retrieve screenshot from Selenium : ".$e->getMessage());
}
```